pkg/extractor/container.go

```diff
@@ -6,16 +6,18 @@ import (
 	"os"
 	"strings"
 
+	"github.com/deepfence/match-scanner/pkg/config"
 	"github.com/deepfence/vessel"
 )
 
 type ContainerExtractor struct {
-	runtime   vessel.Runtime
-	tarReader *tar.Reader
-	rootFile  string
+	runtime     vessel.Runtime
+	tarReader   *tar.Reader
+	rootFile    string
+	matchConfig *config.Config
```

Comment:
Instead of modifying the extractor, what about we filter in the `scanner.go`?
It will drastically simplify the code as we will have 1 place for the filtering shared across all extractors